CHANGELOG entry for gRPC instrumentation (#1307)
newrelic/newrelic-ruby-agent: CHANGELOG.md

  ## v8.10.0

  * **gRPC instrumentation**

>     The agent will now instrument [gRPC](https://grpc.io/) activity performed by clients and servers that use the [grpc](https://rubygems.org/gems/grpc) RubyGem. Instrumentation is automatic and enabled by default, so gRPC users should not need to modify any existing application code or agent configuration to benefit from the instrumentation. The instrumentation makes use of New Relic Distributed Tracing for a comprehensive overview of all gRPC traffic taking place across multiple monitored applications. The Ruby agent's use of Distributed Tracing is compatible with other programming language based New Relic agents that support gRPC.

Reviewer: ```suggestion
    The agent will now instrument [gRPC](https://grpc.io/) activity performed by clients and servers that use the [grpc](https://rubygems.org/gems/grpc) RubyGem. Instrumentation is automatic and enabled by default, so gRPC users should not need to modify any existing application code or agent configuration to benefit from the instrumentation. The instrumentation makes use of New Relic Distributed Tracing for a comprehensive overview of all gRPC traffic taking place across multiple monitored applications. The Ruby agent's use of Distributed Tracing is compatible with other New Relic language agents that support gRPC.
```

Reviewer: I don't think one exists yet, but when a docs page is added for gRPC instrumentation, could you add a link to it in the Changelog?

I think the doc could have more details about external segments being generated by the client, transactions generated by the server and the attributes each publish.

  ## v8.10.0

>   * **gRPC instrumentation**

Reviewer: ```suggestion
  * **New gRPC instrumentation**
```
Take or leave this one!

Author: It reads the same both ways to me. If I see it without the "new", I might think that we've made updates to existing instrumentation, which is not desired. If I see it with the "new", I might think there have been new enhancements added to the existing instrumentation, which is not desired. I'll commit the suggestion to decisively go with one.

>     The agent will now instrument [gRPC](https://grpc.io/) activity performed by clients and servers that use the [grpc](https://rubygems.org/gems/grpc) RubyGem. Instrumentation is automatic and enabled by default, so gRPC users should not need to modify any existing application code or agent configuration to benefit from the instrumentation. The instrumentation makes use of New Relic Distributed Tracing for a comprehensive overview of all gRPC traffic taking place across multiple monitored applications. The Ruby agent's use of Distributed Tracing is compatible with other programming language based New Relic agents that support gRPC.

Reviewer: I'm sorry I didn't catch this last time. Distributed tracing follows a W3C standard, so labeling it as `New Relic` isn't quite appropriate. Infinite tracing, on the other hand, could probably always be prefixed with `New Relic`.
 
```suggestion
    The agent will now instrument [gRPC](https://grpc.io/) activity performed by clients and servers that use the [grpc](https://rubygems.org/gems/grpc) RubyGem. Instrumentation is automatic and enabled by default, so gRPC users should not need to modify any existing application code or agent configuration to benefit from the instrumentation. The instrumentation makes use of distributed tracing for a comprehensive overview of all gRPC traffic taking place across multiple monitored applications. This allows you to observer your client and server activity using any service that adheres to the W3C standard.
```

    The following new configuration parameters have been added for gRPC. All are optional.

    | Configuration name | Default | Behavior |
    | ----------- | ----------- |----------- |
    | `instrumentation.grpc_client` | auto | Set to 'disabled' to disable, set to 'chain' if there are module prepending conflicts |
    | `instrumentation.grpc_server` | auto | Set to 'disabled' to disable, set to 'chain' if there are module prepending conflicts |

>     | `instrumentation.grpc.host_denylist` | "" |  Provide a comma delimited list of host regex patterns (ex: "private.com$,exception.*" to ignore |

Reviewer: ```suggestion
    | `instrumentation.grpc.host_denylist` | "" |  Provide a comma delimited list of host regex patterns (ex: "private.com$,exception.*") |
```
